### ISSUE TITLE
Backpropagation doesn't change weights of the convolutional layer

### ISSUE THREAD

**Issue by sahu-ji** [denfed/leaf-audio-pytorch#3]
Hi, 

Thank you for creating this Github! I was wondering why does forward call of GaborConstraint class has "with torch.no_grad()" (line 28 in leaf_audio_pytorch/convolutions.py)? It appears the convolutional layers weight doesn't change while backpropagation because of this. However, removing it I see that weights update. Is there a reason we need torch.no_grad()? 
Attached a sample code to compare the Gabor layer parameters before and after training a toy code.

`import torch
from leaf_audio_pytorch import frontend
import os
os.environ["CUDA_VISIBLE_DEVICES"] = "0"

device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
a = frontend.Leaf().to(device)

## A random input that is like an audio waveform
x = torch.rand(1,1,97339).cuda(device, non_blocking=True)

a.train()
print(a._complex_conv._kernel.data)
opt = torch.optim.Adam(a.parameters(), lr=0.002, betas=(0.9, 0.999), eps=1e-08, weight_decay=1e-6, amsgrad=True)
for _ in range(100):
    b = a(x)
	
    ## Ground truth for the output for leaf module
    b_gt = torch.ones_like(b)
	
    loss = torch.norm((b-b_gt).view(1,-1),p=2,dim=-1)
    opt.zero_grad()
    loss.backward()
    opt.step()
print('======================================================================================')
print(a._complex_conv._kernel.data)`

**Comment by denfed** [comment]
This is a good catch! I wrote this a little big ago so I forgot the reasoning behind that. Let me do some digging, but it very well might be an error in the code.

**Comment by denfed** [comment]
I cannot think of any reason I put that there. Going to make an update to the code to fix this. Thank you again for the find!
